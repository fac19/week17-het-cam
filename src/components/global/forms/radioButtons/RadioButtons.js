import React from "react";
import RadioButtonFieldset from "./RadioButtons.style";

const RadioButtons = (props) => {
  return (
    <RadioButtonFieldset>
      <legend>{props.question}</legend>
      <label>
        <input
          type="radio"
          name="answer"
          value={props.option1answer}
          required
          onChange={props.onChange}
        />
        {props.option1answer}
      </label>
      <label>
        <input
          type="radio"
          name="answer"
          value={props.option2answer}
          onChange={props.onChange}
        />
        {props.option2answer}
      </label>
    </RadioButtonFieldset>
  );
};

export default RadioButtons;
