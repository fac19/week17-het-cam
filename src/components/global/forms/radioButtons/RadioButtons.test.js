import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import RadioButtons from "./RadioButtons";

test("RadioButtons component renders correctly", () => {
  render(
    <RadioButtons
      question={"What is a tomato?"}
      option1label={"a fruit"}
      option2label={"a dog"}
      option3label={"a pig"}
      option1answer={"a fruit"}
      option2answer={"a dog"}
      option3answer={"a pig"}></RadioButtons>,
  );
  screen.getByText("What is a tomato?");
});

test("RadioButtons component is styled", () => {
  const tree = renderer.create(<RadioButtons />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("display", "flex");
});
