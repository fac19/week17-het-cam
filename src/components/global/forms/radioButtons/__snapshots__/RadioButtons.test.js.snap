// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtons component is styled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > * {
  padding: 1.5rem 0 1.5rem 0;
}

<fieldset
  className="c0"
>
  <legend />
  <label>
    <input
      name="answer"
      required={true}
      type="radio"
    />
  </label>
  <label>
    <input
      name="answer"
      type="radio"
    />
  </label>
</fieldset>
`;
